include/envoy/upstream/upstream.h

```diff
@@ -540,6 +540,8 @@ class PrioritySet {
   COUNTER(update_failure)                                                                          \
   COUNTER(update_no_rebuild)                                                                       \
   COUNTER(update_success)                                                                          \
+  COUNTER(upstream_buffer_overflow_multiple_downstreams_total)                                     \
+  COUNTER(upstream_buffer_overflow_single_downstream_total)                                        \
```

Comment:
What do these counters mean?

It would be good to have some way to provide doc strings for stat counters or at least code comments that explain their meaning.

Reply:
Yes, need to add to the stats [docs](https://github.com/envoyproxy/envoy/blob/ad5dc7db7a4adfb3c0cf299c15f093fe0398a455/docs/root/configuration/upstream/cluster_manager/cluster_stats.rst)
At the moment I've added counters to all places that use the HW/LW watermarks (there may be places that do not need the OW watermark, and so these stats can be removed).

Reply:
These names are confusing to me also. What do they mean exactly?

Reply:
Good to know that some docs exist for other stats.  It would be nice if there was a way to have doc comments here, but that's an issue beyond the scope of your PR.